module/dice/basic-roll.mjs

```diff
@@ -180,4 +180,86 @@ export default class BasicRoll extends Roll {
       return msg.toObject();
     }
   }
+
+  /* -------------------------------------------- */
+  /*  Evaluate Methods                            */
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  async evaluate(options={}) {
+    this.preCalculateDiceTerms(options);
+    return super.evaluate(options);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  evaluateSync(options={}) {
+    this.preCalculateDiceTerms(options);
+    return super.evaluateSync(options);
+  }
+
+  /* -------------------------------------------- */
+  /*  Maximize/Minimize Methods                   */
+  /* -------------------------------------------- */
+
+  /**
+   * Replaces all dice terms that have modifiers with their maximum/minimum value.
+   *
+   * @param {object} [options={}]            Extra optional arguments which describe or modify the BasicRoll.
+   * @returns {RollTerm[]}                   Returns the array of RollTerms.
+   */
+  preCalculateDiceTerms(options={}) {
+    if ( !this._evaluated && ((options.maximize ?? false) || (options.minimize ?? false)) ) {
+      this.terms = this.terms.map(term => {
+        if ( term instanceof foundry.dice.terms.DiceTerm && term.modifiers.length ) {
+          const minimize = ( options.maximize ?? false ) ? false : ( options.minimize ?? false );
```

Comment:
This can be simplified.
```suggestion
    if ( !this._evaluated && (options.maximize || options.minimize) ) {
      this.terms = this.terms.map(term => {
        if ( (term instanceof foundry.dice.terms.DiceTerm) && term.modifiers.length ) {
          const minimize = !options.maximize;
```